core/services/ocr2/plugins/ocr2keeper/evm21/feed_lookup.go

```diff
@@ -182,12 +182,18 @@ func (r *EvmRegistry) doLookup(ctx context.Context, wg *sync.WaitGroup, lookup *
 // allowedToUseMercury retrieves upkeep's administrative offchain config and decode a mercuryEnabled bool to indicate if
 // this upkeep is allowed to use Mercury service.
 func (r *EvmRegistry) allowedToUseMercury(opts *bind.CallOpts, upkeepId *big.Int) (bool, error) {
-	allowed, ok := r.mercury.allowListCache.Get(upkeepId.String())
+	var au activeUpkeep
+	var ok bool
+	if au, ok = r.active[upkeepId.String()]; !ok {
+		return false, fmt.Errorf("upkeep %s is not active", upkeepId)
+	}
+
+	allowed, ok := r.mercury.allowListCache.Get(au.Admin.Hex())
 	if ok {
 		return allowed.(bool), nil
 	}
 
-	cfg, err := r.registry.GetUpkeepPrivilegeConfig(opts, upkeepId)
+	cfg, err := r.registry.GetAdminPrivilegeConfig(opts, au.Admin)
```

Comment:
how does a permission ever expire from the cache? ex if we remove someone's mercury permission...

Reply:
ah I see now, the cache has a cleanup interval. Great solution!